### PR TITLE
CatsSemigroupKForTask added

### PR DESCRIPTION
**Pull request by jendakol**
Solves issue #703

### CODE REVIEW

File: monix-eval/shared/src/main/scala/monix/eval/instances/CatsSemigroupKForTask.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package monix.eval.instances
+import cats.SemigroupK
+import monix.eval.Task
+
+class CatsSemigroupKForTask extends SemigroupK[Task] {
+
+  def combineK[A](a: Task[A], b: Task[A]): Task[A] =
+    Task.catsAsync.handleErrorWith(a)(_ => b)
```
Comment (Avasil) [inline monix-eval/shared/src/main/scala/monix/eval/instances/CatsSemigroupKForTask.scala]: I think we could use `a.onErrorHandleWith(_ => b)` directly but I'm not sure if it makes difference though

**Comment by Avasil** [comment]
Thanks for PR @jendakol -- looks good!

But could you add test in `TypeClassLawsForTaskSuite` to make sure it is and remains lawful instance? :)

**Comment by alexandru** [comment]
Hey, nice.

1. wouldn't this instance make better sense if integrated in `CatsBaseForTask`? https://github.com/monix/monix/blob/master/monix-eval/shared/src/main/scala/monix/eval/instances/CatsBaseForTask.scala
2. if you do it for `Task`, we need it for `Coeval` as well

**Comment by alexandru** [comment]
Also, what @Avasil said, we need to test the laws, so checkout the `TypeClassLaws*` tests.

**Comment by alexandru** [comment]
@jendakol yes, include the instance in [CatsBaseForTask](https://github.com/monix/monix/blob/master/monix-eval/shared/src/main/scala/monix/eval/instances/CatsBaseForTask.scala) for `Task` and in [CatsSyncForCoeval](https://github.com/monix/monix/blob/master/monix-eval/shared/src/main/scala/monix/eval/instances/CatsSyncForCoeval.scala) for `Coeval`.

**Comment by jendakol** [comment]
Updated. Implemented for both `Task` and `Coeval`, enabled laws tests.

**Pull request merged**